tests/functional/cross_db_utils/test_cross_db.py

```diff
@@ -0,0 +1,72 @@
+import os
+import pytest
+from dbt.tests.util import run_dbt
+from tests.functional.cross_db_utils.fixtures import (
+    macros__test_assert_equal_sql,
+    seeds__data_dateadd_csv,
+    models__test_dateadd_sql,
+    models__test_dateadd_yml,
+    seeds__data_datediff_csv,
+    models__test_datediff_sql,
+    models__test_datediff_yml,
+)
+
+
+class BaseCrossDbMacro:
```

Comment:
This class sets up the basic scaffolding for all test cases below. Child methods will inherit methods that don't change (e.g. `packages`, `macros`, `test_build_assert_equal`), and can reimplement others that do (`seeds` + `models`)